<launch>
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  
    <arg name="first_tb3"  default="tb3_0"/>
    <arg name="second_tb3"  default="tb3_1"/>
    <arg name="third_tb3"  default="tb3_2"/>
    <arg name="fourth_tb3"  default="tb3_3"/>
    
    <!-- the position and orientation of the bot1 -->
    <arg name="first_tb3_x_pos" default="0"/>
    <arg name="first_tb3_y_pos" default="0"/>
    <arg name="first_tb3_z_pos" default="0.0"/>
    <arg name="first_tb3_yaw"   default="3.927"/>
    
    <!-- the position and orientation of the bot2 -->
    <arg name="second_tb3_x_pos" default="0"/>
    <arg name="second_tb3_y_pos" default="1"/>
    <arg name="second_tb3_z_pos" default="0.0"/>
    <arg name="second_tb3_yaw"   default="2.365"/>
    
    <!-- the position and orientation of the bot3 -->
    <arg name="third_tb3_x_pos" default="1"/>
    <arg name="third_tb3_y_pos" default="0"/>
    <arg name="third_tb3_z_pos" default="0.0"/>
    <arg name="third_tb3_yaw"   default="5.498"/>
    
    <!-- the position and orientation of the bot4 -->
    <arg name="fourth_tb3_x_pos" default="1"/>
    <arg name="fourth_tb3_y_pos" default="1"/>
    <arg name="fourth_tb3_z_pos" default="0.0"/>
    <arg name="fourth_tb3_yaw"   default="0.785"/>
  
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/empty.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
      </include>
  
    <group ns = "$(arg first_tb3)">
      <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
      <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="publish_frequency" type="double" value="50.0" />
        <param name="tf_prefix" value="$(arg first_tb3)" />
      </node>
      <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model $(arg first_tb3) -x $(arg first_tb3_x_pos) -y $(arg first_tb3_y_pos) -z $(arg first_tb3_z_pos) -Y $(arg first_tb3_yaw) -param robot_description" />
    </group>
  
    <group ns = "$(arg second_tb3)">
       <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
       <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="publish_frequency" type="double" value="50.0" />
        <param name="tf_prefix" value="$(arg second_tb3)" />
      </node>
       <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model $(arg second_tb3) -x $(arg second_tb3_x_pos) -y $(arg second_tb3_y_pos) -z $(arg second_tb3_z_pos) -Y $(arg second_tb3_yaw) -param robot_description" />
    </group>
  
    <group ns = "$(arg third_tb3)">
      <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
      <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="publish_frequency" type="double" value="50.0" />
        <param name="tf_prefix" value="$(arg third_tb3)" />
      </node>
       <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model $(arg third_tb3) -x $(arg third_tb3_x_pos) -y $(arg third_tb3_y_pos) -z $(arg third_tb3_z_pos) -Y $(arg third_tb3_yaw) -param robot_description" />
    </group>
    
    <group ns = "$(arg fourth_tb3)">
      <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
      <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="publish_frequency" type="double" value="50.0" />
        <param name="tf_prefix" value="$(arg fourth_tb3)" />
      </node>
       <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model $(arg fourth_tb3) -x $(arg fourth_tb3_x_pos) -y $(arg fourth_tb3_y_pos) -z $(arg fourth_tb3_z_pos) -Y $(arg fourth_tb3_yaw) -param robot_description" />
    </group>

    
    
  </launch>